crates/ruff_linter/src/rules/ruff/rules/parenthesize_logical_operators.rs

```diff
@@ -75,18 +83,30 @@ pub(crate) fn parenthesize_chained_logical_operators(
                     ..
                 },
             ) => {
+                let locator = checker.locator();
+                let source_range = bool_op.range();
                 if parenthesized_range(
                     bool_op.into(),
                     expr.into(),
                     checker.indexer().comment_ranges(),
-                    checker.locator().contents(),
+                    locator.contents(),
                 )
                 .is_none()
                 {
-                    checker.diagnostics.push(Diagnostic::new(
-                        ParenthesizeChainedOperators,
-                        bool_op.range(),
-                    ));
+                    let mut diagnostic =
+                        Diagnostic::new(ParenthesizeChainedOperators, source_range);
+                    let line_length: u16 = checker.settings.line_length.into();
+                    if !locator.contains_line_break(source_range)
+                        && locator.full_line(source_range.start()).trim_end().len()
+                            <= ((line_length as usize) - 2)
+                    {
+                        let new_source = format!("({})", locator.slice(source_range));
+                        diagnostic.set_fix(Fix::applicable_edit(
+                            Edit::range_replacement(new_source, source_range),
+                            Applicability::Safe,
```

Comment:
We're lucky that Python doesn't support overloading the `and` and `or` operators.